### ISSUE TITLE
Icon nomenclature

### ISSUE THREAD

**Issue by meetdilip** [phw/peek#123]
I had my own Peek icon added to Papirus Icon theme for 16.04. But, with the latest update, in usr/share/applications peek.desktop file got overwritten and the icon was again changed to com.~lobster.peek.

It is easier to maintain with icon name as either **peek** or **peek.svg** to work with icon themes. Kindly consider following the same with icon names. Thanks.

**Comment by phw** [comment]
No, won't change it, sorry. The fully namespaced icon name is a requirement for icons exported byFlatpak, and I don't see a reason to handle the naming differently in other cases. How exactly does this make problems with icon themes? You should be able to just use `com.uploadedlobster.peek.png` as a name for custom icons as well.

**Comment by meetdilip** [comment]
Is it possible have a name for icon that starts with **peek**  ? Would be easier to manage that way. 

PS : In Papirus, most icon names are just their app names. I am not sure why it is different in this case.

**Comment by phw** [comment]
The icon name has to start with com.uploadedlobster.peek, after that it doesn't matter.

Traditionally the icon names were the same as the executable, but the trend is to have all those files installed in shared location have to use the full namespaced app name to avoid clashes.

I still don't understand how having the icon names start with peek would make any difference for icon themes. AFAIK the names in a theme just have to match. If an icon is present in a theme that is used, if not it searches in base themes or in hicolor as final fallback.

**Comment by meetdilip** [comment]
Just trying to explain my point. Please see how icons are named in the apps folder of Papirus

![papirus](https://cloud.githubusercontent.com/assets/6965443/24426086/05292522-1424-11e7-865b-9f991f07901a.png)

**Comment by phw** [comment]
Sorry for the delay. I am quite aware that the majority of applications use the executable name for the icon as well, that has been that way for a long time and even was the recommendation in the desktop file spec. But applications are moving towards namespaced resource names (for desktop files, DBus service files, appdata descriptions, geschema files and icons), and Flatpak even makes this a requirement. Most Gnome apps already do this. Have a look what `ls /usr/share/icons/hicolor/32x32/apps/*.*.*.*` will output on your system. If you have some Gnome apps installed chances are there are a few namespaced icons as well.

If you use com.uploadedlobster.peek.png in the icon theme that should just work as well. Yes, the icon name is a bit longer, but this also makes it unique in case some other application also gets called Peek.

In short: If at all this is only a tiny issue, and renaming the icon is not an option anyway.
